src/main/java/com/spikes2212/path/OdometryHandler.java

```diff
@@ -0,0 +1,177 @@
+package com.spikes2212.path;
+
+import java.awt.geom.Point2D;
+import java.util.function.Supplier;
+
+/**
+ *<ul><i><a><h1><li><strong><b>The calculate method should be called repeatedly, otherwise your robot's position
+ * 	would not be recalculated, and therefore will be substantially inaccurate.
+ * </b></strong></li></h1></a></i></ul>
+ *
+ * a class which uses encoders to find the progress of a robot between discrete
+ * times
+ *
+ * @author noam mantin
+ *
+ */
+public class OdometryHandler {
+
+    private RelativeDataSupplier yawDiff, leftDistance, rightDistance;
+    private Supplier<Double> angleSupplier;
+    private double x, y;
+
+    /**
+     * creates a new {@link OdometryHandler} object, with given parameters
+     * @param leftPosition  the left position supplier
+     * @param rightPosition the right position supplier
+     * @param angleSupplier the angle supplier
+     * @param x the starting x coordinate of your robot
+     * @param y the starting y coordinate of your robot
+     * @param baseYaw the angle between the x axis and your robot's starting position
+     */
+    public OdometryHandler(Supplier<Double> leftPosition, Supplier<Double> rightPosition, Supplier<Double> angleSupplier
+            , double x, double y, double baseYaw) {
+        this.leftDistance = new RelativeDataSupplier(leftPosition);
+        this.rightDistance = new RelativeDataSupplier(rightPosition);
+        this.yawDiff = new RelativeDataSupplier(angleSupplier);
+        this.angleSupplier = angleSupplier;
+        this.x = x;
+        this.y = y;
+    }
+
+    /**
+     * <ul><i><a><h1><li><strong><b>This method should be called repeatedly, otherwise your robot's position
+     * 	would not be recalculated, and therefore will be substantially inaccurate.
+     * </b></strong></li></h1></a></i></ul>
+     *
+     * Calculates and returns the displacement of the robot since the last
+     * check.</br>
+     *
+     * for more information about this process, see pages 2910-2911 in the main
+     * article from the README file
+     *
+     * @return the position rate vector of the robot (ΔX and ΔY), from the
+     *         last call to this method, in the navigation frame.
+     *
+     */
+    public void calculate() {
```

Comment:
make `calculate` return a `Waypoint` which holds the current location of the robot